File: lib/services/executors/claude-agent/src/cli-config-error.ts
// Copyright © 2026 Fall Guy Consulting.
// Licensed under the Apache License, Version 2.0.
// See LICENSE.apache at the repo root.

export class CliConfigError extends Error {
  readonly errorClass = "agent/attribute_invalid";

  constructor(message: string) {
    super(message);
    this.name = "CliConfigError";
  }
}

export function isCliConfigError(e: unknown): e is CliConfigError {
  return e instanceof CliConfigError;
}

File: lib/services/executors/claude-agent/cliconfigerror.go
// Copyright © 2026 Fall Guy Consulting.
// SPDX-License-Identifier: AGPL-3.0-or-later OR LicenseRef-FallGuy-Commercial

package claudeagent

type CliConfigError struct {
	Message string
}

func (e *CliConfigError) Error() string { return e.Message }

func (e *CliConfigError) ErrorClass() string { return "agent/attribute_invalid" }
